### PR TITLE
make generated OpenSplice code reproducible

### PR DESCRIPTION
**Pull request by dirk-thomas**
* http://ci.ros2.org/job/ci_linux/1118/
* http://ci.ros2.org/job/ci_osx/902/
* http://ci.ros2.org/job/ci_windows/1181/

### CODE REVIEW

**Comment by jacquelinekay** [comment]
I tested this with std_msgs. This is an example header (to `Header_DCPS.cpp`)

```
//******************************************************************
// 
//  Generated by IDL to C++ Translator
//  
//  File name: Header_Dcps.cpp
//  Source: Header_
//  Generated: timestamp removed to make the build reproducible
//  OpenSplice V6.4.140407OSS
//  
//******************************************************************
```

`Source: Header_` looks a little funny to me, should that be `Header_DCPS.idl`? Or should it have a message similar to the `Generated` line?

File: rosidl_typesupport_opensplice_cpp/rosidl_typesupport_opensplice_cpp/__init__.py [hunk 1]
```diff
@@ -102,6 +116,23 @@ def _copy_constructor_and_assignment_operator(lines, msg_name):
     assert False, 'Failed to find insertion point'
 
 
+def _replace_path_and_timestamp(lines, msg_name, idl_path):
+    found_source = False
+    for i, line in enumerate(lines):
+        if line.startswith('//  Source: '):
+            assert not found_source, "More than one '// Source: ' line was found"
+            found_source = True
+            lines[i] = '//  Source: ' + msg_name
```
Comment (dirk-thomas) [inline rosidl_typesupport_opensplice_cpp/rosidl_typesupport_opensplice_cpp/__init__.py]: This should be `idl_path` - not `msg_name`.

**Comment by dirk-thomas** [comment]
Ups, I fixed it. It is supposed to contain `<PKGNAME>/msg/dds_opensplice/<MSGNAME>_.idl`.

**Comment by jacquelinekay** [comment]
looks good now. +1

**Pull request merged**